runtime/cudaq/dynamics/cudm_helpers.cpp

```diff
@@ -0,0 +1,209 @@
+    for (size_t i = 0; i < dim; i++) {
+      for (size_t j = 0; j < dim; j++) {
+        flat_matrix[i * dim + j] = c_op[{i, j}];
+      }
+    }
+
+    // Create Operator term for LtL and add to lindblad_op
+    cudensitymatOperatorTerm_t term;
+    HANDLE_CUDM_ERROR(cudensitymatCreateOperatorTerm(
+        handle, static_cast<int32_t>(mode_extents.size()), mode_extents.data(),
+        &term));
+    cudensitymatDestroyOperator(lindblad_op);
+
+    // Attach terms and cleanup
+    cudensitymatWrappedScalarCallback_t scalarCallback = {nullptr, nullptr};
+    HANDLE_CUDM_ERROR(cudensitymatOperatorAppendTerm(handle, lindblad_op, term,
+                                                     0, {1.0}, scalarCallback));
+    cudensitymatDestroyOperatorTerm(term);
+  }
+
+  return lindblad_op;
+}
+
+std::map<int, int>
+convert_dimensions(const std::vector<int64_t> &mode_extents) {
+  std::map<int, int> dimensions;
+  for (size_t i = 0; i < mode_extents.size(); i++) {
+    dimensions[static_cast<int>(i)] = static_cast<int>(mode_extents[i]);
+  }
+  return dimensions;
+}
+
+cudensitymatOperator_t convert_to_cudensitymat_operator(
+    cudensitymatHandle_t handle,
+    const std::map<std::string, std::complex<double>> &parameters,
+    const operator_sum &op, const std::vector<int64_t> &mode_extents) {
+  try {
+    cudensitymatOperator_t operator_handle;
+    auto status = cudensitymatCreateOperator(
+        handle, static_cast<int32_t>(mode_extents.size()), mode_extents.data(),
+        &operator_handle);
+    if (status != CUDENSITYMAT_STATUS_SUCCESS) {
+      throw std::runtime_error("Failed to create operator.");
+    }
+
+    for (const auto &product_op : op.get_terms()) {
+      cudensitymatOperatorTerm_t term;
+
+      HANDLE_CUDM_ERROR(cudensitymatCreateOperatorTerm(
+          handle, static_cast<int32_t>(mode_extents.size()),
+          mode_extents.data(), &term));
+
+      for (const auto &component : product_op.get_terms()) {
+        if (std::holds_alternative<cudaq::elementary_operator>(component)) {
+          const auto &elem_op = std::get<cudaq::elementary_operator>(component);
+
+          // Create a cudensitymat elementary operator
+          cudensitymatElementaryOperator_t cudm_elem_op;
+
+          // Get the matrix representation of elementary operator
+          auto dimensions = convert_dimensions(mode_extents);
+          auto matrix = elem_op.to_matrix(dimensions, parameters);
+
+          // Flatten the matrix into a single-dimensional array
+          std::vector<std::complex<double>> flat_matrix;
+          for (size_t i = 0; i < matrix.get_rows(); i++) {
+            for (size_t j = 0; j < matrix.get_columns(); j++) {
+              flat_matrix.push_back(matrix[{i, j}]);
+            }
+          }
+
+          // Create a cudensitymat elementary operator
+          HANDLE_CUDM_ERROR(cudensitymatCreateElementaryOperator(
+              handle, 1, mode_extents.data(),
```

Comment:
Q: is the `mode_extents` here all the extents? 
If so, we need to retrieve the extent of the sub-space this elementary operation acts on.